# SPDX-License-Identifier: Apache-2.0 OR ISC OR MIT
diff --git a/sw/device/tests/crypto/mlkem_functest.c b/sw/device/tests/crypto/mlkem_functest.c
--- a/sw/device/tests/crypto/mlkem_functest.c
+++ b/sw/device/tests/crypto/mlkem_functest.c
@@ -312,6 +312,7 @@ static void test_mlkem1024_derand(void) {
                   kOtcryptoMlkem1024SharedSecretBytes);
 }
 
+#if 0
 static void test_mlkem512_randomized(void) {
   uint64_t t0;
 
@@ -492,6 +493,8 @@ static void test_mlkem1024_randomized(void) {
   CHECK_ARRAYS_EQ(key_a_blob, key_b_blob, ARRAYSIZE(key_a_blob));
 }
 
+#endif
+
 bool test_main(void) {
   CHECK_STATUS_OK(entropy_testutils_auto_mode_init());
 
@@ -500,10 +503,5 @@ bool test_main(void) {
   test_mlkem768_derand();
   test_mlkem1024_derand();
 
-  // Run randomized tests
-  test_mlkem512_randomized();
-  test_mlkem768_randomized();
-  test_mlkem1024_randomized();
-
   return true;
 }
