# SPDX-License-Identifier: Apache-2.0 OR ISC OR MIT
diff --git a/sw/device/tests/crypto/mlkem_functest.c b/sw/device/tests/crypto/mlkem_functest.c
--- a/sw/device/tests/crypto/mlkem_functest.c
+++ b/sw/device/tests/crypto/mlkem_functest.c
@@ -355,6 +355,7 @@ static void test_mlkem1024_derand(void) {
                             expected_key, kOtcryptoMlkem1024SharedSecretBytes);
 }
 
+#if 0
 static void test_mlkem512_randomized(void) {
   uint64_t t0;
 
@@ -538,6 +539,8 @@ static void test_mlkem1024_randomized(void) {
                             kOtcryptoMlkem1024SharedSecretBytes);
 }
 
+#endif
+
 bool test_main(void) {
   CHECK_STATUS_OK(entropy_testutils_auto_mode_init());
 
@@ -546,10 +549,5 @@ bool test_main(void) {
   test_mlkem768_derand();
   test_mlkem1024_derand();
 
-  // Run randomized tests
-  test_mlkem512_randomized();
-  test_mlkem768_randomized();
-  test_mlkem1024_randomized();
-
   return true;
 }
