diff --git a/crypto/fipsmodule/ml_kem/mlkem_native_config.h b/crypto/fipsmodule/ml_kem/mlkem_native_config.h
--- a/crypto/fipsmodule/ml_kem/mlkem_native_config.h
+++ b/crypto/fipsmodule/ml_kem/mlkem_native_config.h
@@ -20,8 +20,14 @@

 // Everything is built in a single CU, so both internal and external
 // mlkem-native API can have internal linkage.
+// Mark as unused to suppress warnings for functions not used by AWS-LC
+#if defined(__GNUC__) || defined(__clang__)
+#define MLK_CONFIG_INTERNAL_API_QUALIFIER static __attribute__((unused))
+#define MLK_CONFIG_EXTERNAL_API_QUALIFIER static __attribute__((unused))
+#else
 #define MLK_CONFIG_INTERNAL_API_QUALIFIER static
 #define MLK_CONFIG_EXTERNAL_API_QUALIFIER static
+#endif

 // Enable PCT if and only if AWS-LC is built in FIPS-mode.
 #if defined(AWSLC_FIPS)
@@ -78,8 +84,9 @@ static MLK_INLINE void mlk_zeroize(void *ptr, size_t len) {
 #include <stdint.h>
 #include "mlkem/sys.h"
 #include <openssl/rand.h>
-static MLK_INLINE void mlk_randombytes(void *ptr, size_t len) {
-    AWSLC_ABORT_IF_NOT_ONE(RAND_bytes(ptr, len));
+static MLK_INLINE int mlk_randombytes(void *ptr, size_t len) {
+  AWSLC_ABORT_IF_NOT_ONE(RAND_bytes(ptr, len));
+  return 0;
 }
 #endif // !__ASSEMBLER__
